precompile/contracts/warp/README.md

```diff
@@ -77,27 +77,35 @@ Since the predicate is encoded into the [Transaction Access List](https://eips.e
 
 Therefore, we use the [Predicate Utils](https://github.com/ava-labs/coreth/blob/master/predicate/Predicate.md) package to encode the actual byte slice of size N into the access list.
 
-### Performance Optimization: C-Chain to Avalanche L1
+### Performance Optimization: Primary Network to Avalanche L1
 
-For communication between the C-Chain and an L1, as well as broader interactions between the Primary Network and Avalanche L1s, we have implemented special handling for the C-Chain.
+The Primary Network has a large validator set compared to most Subnets and L1s, which makes Warp signature collection and verification from the entire Primary Network validator set costly. All Subnets and L1s track at least one blockchain of the Primary Network, so we can instead optimize this by using the validator set of the receiving L1 instead of the Primary Network for certain Warp messages.
 
-The Primary Network has a large validator set, which creates a unique challenge for Avalanche Warp messages. To reach the required stake threshold, numerous signatures would need to be collected and verifying messages from the Primary Network would be computationally costly. However, we have developed a more efficient solution.
+#### Subnets
 
-When an Avalanche L1 receives a message from a blockchain on the Primary Network, we use the validator set of the receiving L1 instead of the entire network when validating the message. Note this is NOT possible if an L1 does not validate the Primary Network, in which case the Warp precompile must be configured with `requirePrimaryNetworkSigners`.
+Recall that Avalanche Subnet validators must also validate the Primary Network, so it tracks all of the blockchains in the Primary Network (X, C, and P-Chains).
 
-Sending messages from the C-Chain remains unchanged.
-However, when L1 XYZ receives a message from the C-Chain, it changes the semantics to the following:
+When an Avalanche Subnet receives a message from a blockchain on the Primary Network, we use the validator set of the receiving Subnet instead of the entire network when validating the message. 
 
-1. Read the `SourceChainID` of the signed message (C-Chain)
-2. Look up the `SubnetID` that validates C-Chain: Primary Network
-3. Look up the validator set of L1 XYZ (instead of the Primary Network) and the registered BLS Public Keys of L1 XYZ at the P-Chain height specified by the ProposerVM header
-4. Continue Warp Message verification using the validator set of L1 XYZ instead of the Primary Network
+Sending messages from the X, C, or P-Chain remains unchanged.
+However, when the Subnet receives the message, it changes the semantics to the following:
 
-This means that C-Chain to L1 communication only requires a threshold of stake on the receiving L1 to sign the message instead of a threshold of stake for the entire Primary Network.
+1. Read the `SourceChainID` of the signed message
+2. Look up the `SubnetID` that validates `SourceChainID`: Primary Network
```

Comment:
nit maybe that's a bit confusing, it being previously

> Look up the `SubnetID` that validates C-Chain: Primary Network

Maybe this would be better?

```suggestion
2. Look up the `SubnetID` that validates `SourceChainID: Primary Network`
```

? I'm not 100% sure on the original documentation meaning either, so maybe I'm missing out information too.